client/driver/docker.go

```diff
@@ -1368,22 +1368,17 @@ CREATE:
 // transient Docker errors.
 func (d *DockerDriver) startContainer(c *docker.Container) error {
 	// Start a container
-	attempted := 0
-START:
 	startErr := client.StartContainer(c.ID, c.HostConfig)
 	if startErr == nil {
 		return nil
 	}
 
-	d.logger.Printf("[DEBUG] driver.docker: failed to start container %q (attempt %d): %v", c.ID, attempted+1, startErr)
+	d.logger.Printf("[DEBUG] driver.docker: failed to start container %q: %v", c.ID, startErr)
 
 	// If it is a 500 error it is likely we can retry and be successful
+	// Allow recovery parameters to handle the retries
 	if strings.Contains(startErr.Error(), "API error (500)") {
-		if attempted < 5 {
-			attempted++
-			time.Sleep(1 * time.Second)
-			goto START
-		}
+		return structs.NewRecoverableError(startErr, true)
```

Comment:
Can you change it such that we still retry quickly for 5 times and then fall back to the recoverable error if it is a 500? We want the fast retry behavior in this case since often it recovers quickly and the users restart policy may cause the task to fail.